### PR TITLE
only scan exploitable wordpress things

### PR DESCRIPTION
**Pull request by h00die**
This PR does a few things:

1. `rubocop -a` on `wordpress_scanner.rb`
2. creates a new option in `wordpress_scanner` called `exploitable`.  If its set (default true), it only scans for plugins and themes which MSF has exploits for.
3. creates a new script `tools/dev/update_wordpress_vulnerabilities.rb` which scans all exploits and aux folders for ruby files, then checks if they call the `check_plugin_version_from_readme` or `check_theme_version_from_readme` function.  If they do, pull the name.  Finally, write it in to the `wordpress_scanner.rb`
4. update wordpress plugin/theme files

The list of wordpress things is just getting overwhelming, and while you may want to check for all 90,563 plugins, it's slightly faster to search for just the ~40 there are actual modules for.  Same thing for themes. 

This is NOT a perfect system, it is very likely in the future (or its happened already) some modules will NOT call the scanned for function.  In which case, I say just put it in there as a comment so the hit will still happen.  Ref: https://github.com/rapid7/metasploit-framework/pull/15402#issuecomment-878156409

However, I think it's a pretty decent stab at automating a way to only scan for the .05% things we are most likely to care about.
The other potential option is just making `data/wordlists/wp-[item]-vuln.txt` and writing to there instead of the module itself, but if we have to write somewhere, and the list isn't long, I figured write to the module itself to avoid read the file at runtime

## Verification

- [ ] Start `msfconsole`
- [ ] `use auxiliary/scanner/http/wordpress_scanner`
- [ ] `set exploitable true`
- [ ] `set rhosts [ip]`
- [ ] **Verify** it only scans vulnerable things
- [ ] `set exploitable false`
- [ ] **Verify** it scans all the things
- [ ] Run  `tools/dev/update_wordpress_vulnerabilities.rb`
- [ ] **Verify** it updates `auxiliary/scanner/http/wordpress_scanner.rb`

### CODE REVIEW

File: modules/auxiliary/scanner/http/wordpress_scanner.rb [hunk 1]
```diff
@@ -33,6 +34,16 @@ def initialize
     ]
   end
 
+  def vuln_themes
+    vuln_themes = ['holding_pattern', 'wplms']
+    return vuln_themes
+  end
+
+  def vuln_plugins
+    vuln_plugins = ['backup', 'modern-events-calendar-lite', 'ninja-forms', 'simple-file-list', 'sp-client-document-manager', 'drag-and-drop-multiple-file-upload-contact-form-7', 'wp-file-manager', 'work-the-flow-file-upload', 'ajax-load-more', 'wpdiscuz', 'wptouch', 'front-end-editor', 'wpshop', 'plainview-activity-monitor', 'sexy-contact-form', 'download-manager', 'inboundio-marketing', 'wp-mobile-detector', 'website-contact-form-with-file-upload', 'slideshow-gallery', 'reflex-gallery', 'wp-symposium', 'photo-gallery', 'wysija-newsletters', 'all-in-one-wp-migration', 'wp-ultimate-csv-importer', 'wp-symposium', 'wp-gdpr-compliance', 'wp-easycart', 'dukapress', 'loginizer', 'email-subscribers', 'wordpress-mobile-pack', 'wp-mobile-edition', 'boldgrid-backup', 'gi-media-library', 'chopslider', 'nextgen-gallery', 'simple-backup', 'subscribe-to-comments', 'easy-wp-smtp', 'duplicator_download', 'woocommerce-abandoned-cart']
```
[the remark this answers is not in the document]
Comment (h00die) [reply modules/auxiliary/scanner/http/wordpress_scanner.rb]: 10-4, adjusted to write to text files instead

**Review by jheysel-r7 — CHANGES_REQUESTED** [review]
Great PR, thanks for all the effort on this h00die! Noticed a couple typos but other than that it looks good. 

I was able to recreate the test environment without issue. Set up docksal, installed the plugin `drag-and-drop-multiple-file-upload-contact-form-7` which we have a module for, set `EXPLOITABLE` to `true` and watched the module run (efficiently) and identify the wp-plugin we have a module for:
```
msf6 > use scanner/http/wordpress_scanner

Matching Modules
================

   #  Name                                      Disclosure Date  Rank    Check  Description
   -  ----                                      ---------------  ----    -----  -----------
   0  auxiliary/scanner/http/wordpress_scanner                   normal  No     Wordpress Scanner


Interact with a module by name or index. For example info 0, use 0 or use auxiliary/scanner/http/wordpress_scanner

[*] Using auxiliary/scanner/http/wordpress_scanner
msf6 auxiliary(scanner/http/wordpress_scanner) > options

Module options (auxiliary/scanner/http/wordpress_scanner):

   Name                 Current Setting                                      Required  Description
   ----                 ---------------                                      --------  -----------
   EXPLOITABLE          true                                                 no        Only scan plugins and themes which a MSF module exists for
   EXPLOITABLE_PLUGINS  /Users/jheysel/rapid7/metasploit-framework/data/wor  yes       File containing exploitable by MSF plugins
                        dlists/wp-exploitable-plugins.txt
   EXPLOITABLE_THEMES   /Users/jheysel/rapid7/metasploit-framework/data/wor  yes       File containing exploitable by MSF themes
                        dlists/wp-exploitable-themes.txt
   PLUGINS              true                                                 no        Detect plugins
   PLUGINS_FILE         /Users/jheysel/rapid7/metasploit-framework/data/wor  yes       File containing plugins to enumerate
                        dlists/wp-plugins.txt
   PROGRESS             1000                                                 yes       how often to print progress
   Proxies                                                                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS                                                                    yes       The target host(s), see https://github.com/rapid7/metasploit-framework/wiki/Using-Metasploit
   RPORT                80                                                   yes       The target port (TCP)
   SSL                  false                                                no        Negotiate SSL/TLS for outgoing connections
   TARGETURI            /                                                    yes       The base path to the wordpress application
   THEMES               true                                                 no        Detect themes
   THEMES_FILE          /Users/jheysel/rapid7/metasploit-framework/data/wor  yes       File containing themes to enumerate
                        dlists/wp-themes.txt
   THREADS              1                                                    yes       The number of concurrent threads (max one per host)
   VHOST                                                                     no        HTTP server virtual host

msf6 auxiliary(scanner/http/wordpress_scanner) > set vhost msf-wp.docksal.site
vhost => msf-wp.docksal.site
msf6 auxiliary(scanner/http/wordpress_scanner) > set rhosts 192.168.123.1
rhosts => 192.168.123.1
msf6 auxiliary(scanner/http/wordpress_scanner) > run

[*] Trying 192.168.123.1
[+] 192.168.123.1 - Detected Wordpress 5.3
[*] 192.168.123.1 - Enumerating Themes
[*] 192.168.123.1 - Progress  0/2 (0.0%)
[*] 192.168.123.1 - Finished scanning themes
[*] 192.168.123.1 - Enumerating plugins
[*] 192.168.123.1 - Progress   0/44 (0.0%)
[+] 192.168.123.1 - Detected plugin: drag-and-drop-multiple-file-upload-contact-form-7 version 1.3.6.1
[*] 192.168.123.1 - Finished scanning plugins
[*] 192.168.123.1 - Finished all scans
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

To test `update_wordpress_vulnerabilities.rb` I installed a second wp-plugin `wp-test-email` that we didn't have a module for. I created a dummy module for it:
Screen Shot 2021-09-21 at 1 33 53 PM

Ran the update script, ensured that it updated `wp-exploitable-plugins.txt` and then reran the module to ensure it identified the new wp-plugin:
```
msf6 auxiliary(scanner/http/wordpress_scanner) > run

[*] Trying 192.168.123.1
[+] 192.168.123.1 - Detected Wordpress 5.3
[*] 192.168.123.1 - Enumerating Themes
[*] 192.168.123.1 - Progress  0/2 (0.0%)
[*] 192.168.123.1 - Finished scanning themes
[*] 192.168.123.1 - Enumerating plugins
[*] 192.168.123.1 - Progress   0/45 (0.0%)
[+] 192.168.123.1 - Detected plugin: drag-and-drop-multiple-file-upload-contact-form-7 version 1.3.6.1
[+] 192.168.123.1 - Detected plugin: wp-test-email version 1.1.5
[*] 192.168.123.1 - Finished scanning plugins
[*] 192.168.123.1 - Finished all scans
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

@h00die Just a couple questions, excuse my ignorance, but will the update script be executed on some regular cadence or will it need to be run manually?

Also I noticed the file permissions on  `update_wordpress_vulnerabilities.rb` were set to 644 when the other files in side `metasploit-framework/tools/dev/` are set to 755. Didn't seem to affect the module but should that be changed for consistency?

**Comment by h00die** [comment]
> 
> @h00die Just a couple questions, excuse my ignorance, but will the update script be executed on some regular cadence or will it need to be run manually?
> 

It'll need to be run manually.  However, it isn't very often that new wordpress modules are added, so running it with the PR would most likely be the best.

**Review by jheysel-r7 — APPROVED** [review]

**Comment by jheysel-r7** [comment]
# Release Notes

This adds options to the wordpress_scanner that enables the user to only scan for wordpress themes or plugins that metasploit has modules for.

**Pull request merged**